A32-xds-circuit-breaking.md

```diff
@@ -0,0 +1,157 @@
+# gRPC xDS circuit breaking
+
+* Author(s): Chengyuan Zhang (voidzcy)
+* Approver: markdroth
+* Status: In Review
+* Implemented in:
+* Last updated: 2020-07-20
+* Discussion at: https://groups.google.com/g/grpc-io/c/NEx70p8mcjg
+
+
+## Abstract
+
+Adopt Envoy's [circuit breaking](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/circuit_breaking#circuit-breaking) 
+component that comes with the xDS protocol to
+fail network operations quickly and apply back pressure downstream as soon as
+possible. This allows gRPC clients to received distributed traffic limit 
+configurations that prevent cascaded service overload failures. 
+
+## Background
+
+Envoy supports configurable circuit breaking settings on per upstream cluster 
+and per priority basis to allow different components of the distributed system
+to be tuned independently. Since gRPC xDS does not support [priority based
+routing](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/http_routing#arch-overview-http-routing-priority), 
+requests to a single upstream cluster conform to the single circuit breaking
+limit.
+
+Envoy handles HTTP traffic with [connection pooling](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/connection_pooling#connection-pooling),
+many circuit breaking limits are configured for connection/connection pool 
+management. Since each gRPC client creates at most one HTTP/2 connection to 
+a single upstream endpoint address, those configurations do not apply to gRPC
+and will be ignored. See more in [Other configurations considered](#other-configurations-considered).
+
+As per upstream cluster configurations, circuit breakers will be implemented as
```

Comment:
Suggest writing this paragraph as something like this:

With the introduction of [gRFC A28: xDS Traffic Splitting and Routing](https://github.com/grpc/proposal/blob/master/A28-xds-traffic-splitting-and-routing.md), it became possible for there to be more than one CDS LB policy instance for a given cluster (e.g., if one route pointed to a given cluster and a different route split traffic between that cluster and other clusters).  However, after implementing the changes described in [gRPC A31: xDS RouteActions Support](https://github.com/grpc/proposal/pull/192), there will be only one CDS LB policy instance for a given cluster, which will make the CDS LB policy the logical place to implement circuit breaking.

Reply:
Rephrased.